# HG changeset patch
# User Dan Villiom Podlaski Christiansen <danchr@gmail.com>
# Date 1747228504 -7200
#      Wed May 14 15:15:04 2025 +0200
# Branch stable
# Node ID 2f32cc33c8c68d1a20bd8427e293433e5d15b8cd
# Parent  809a4c4725436a6f3755e58cb864faed4264ad25
# EXP-Topic patches
allow_rust

diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -495,13 +495,7 @@ class hgbuildpy(build_py):
 
     def run(self):
         rust = self.distribution.rust
-        if self.distribution.pure:
-            modulepolicy = 'py'
-        elif self.build_lib == '.':
-            # in-place build should run without rebuilding and Rust extensions
-            modulepolicy = 'rust+c-allow' if rust else 'allow'
-        else:
-            modulepolicy = 'rust+c' if rust else 'c'
+        modulepolicy = 'rust+c-allow'
 
         content = b''.join(
             [
